mutex.go

```diff
@@ -0,0 +1,33 @@
+package mutexasserts
+
+import (
+	"sync"
+)
+
+func MutexLocked(m *sync.Mutex) bool {
+	ok := m.TryLock()
+	if ok {
+		m.Unlock()
+	}
+	return !ok
+}
+
+func RWMutexLocked(rw *sync.RWMutex) bool {
+	rlockOk := rw.TryRLock()
+	if rlockOk {
+		rw.RUnlock()
+	}
+	return !rlockOk
+}
+
+func RWMutexRLocked(rw *sync.RWMutex) bool {
+	lockedOk := rw.TryLock()
+	if lockedOk {
+		rw.Unlock()
+	}
+	rlockedOk := rw.TryRLock()
```

Comment:
`sync.RWMutex` is single-writer and multi-reader. So `TryRLock` would report success even though the lock wasn't locked previously.